src/server/verification/processor/provider/ZoomProvider.js

```diff
@@ -117,47 +171,34 @@ class ZoomProvider implements IEnrollmentProvider {
         throw exception
       }
 
-      // if exception has response checking
-      // is subCode non-equals to 'nameCollision'
-      // that we have some enrollment error
-      // (e.g. liveness wasn't passsed, glasses detected, poor quality)
-      if ('nameCollision' !== response.subCode) {
-        const isEnrolled = false
-        const isLive = api.isLivenessCheckPassed(response)
-
-        // then notifying & throwing enrollment exception
-        await notifyProcessor({ isEnrolled, isLive })
-        throwCustomException(message, { isEnrolled, isLive }, response)
-      }
-
-      // otherwise, if subCode equals to 'nameCollision'
-      // that means identifier was already enrolled
-      // as we've already passed dupliucate check
-      // we don't throw anything, but setting alreadyEnrolled flag
-      alreadyEnrolled = true
+      // otherwise notifying & throwing enrollment exception
+      isEnrolled = false
 
-      // returning 'already enrolled' status
-      enrollmentStatus = 'The FaceMap was already enrolled.'
+      await notifyProcessor({ isEnrolled })
+      throwCustomException(message, { isEnrolled }, response)
     }
 
+    // preparing corresponding success message depinding of the alreadyEnrolled status
+    const enrollmentStatus = alreadyEnrolled ? alreadyEnrolledMessage : successfullyEnrolledMessage
+
     // notifying about successfull enrollment
-    await notifyProcessor({ isEnrolled: true, isLive: true })
+    await notifyProcessor({ isEnrolled })
+
     // returning successfull result
     return { isVerified: true, alreadyEnrolled, message: enrollmentStatus }
   }
 
-  async enrollmentExists(enrollmentIdentifier: string, customLogger = null): Promise<boolean> {
-    const { api, logger } = this
+  async isEnrollmentIndexed(enrollmentIdentifier: string, customLogger = null): Promise<boolean> {
+    const { api, logger, _isNotExistsException } = this
     const log = customLogger || logger
+    const { defaultSearchIndexName } = api
 
     try {
-      await api.readEnrollment(enrollmentIdentifier, customLogger)
+      await api.readEnrollmentIndex(enrollmentIdentifier, defaultSearchIndexName, customLogger)
```

Comment:
why did we change it? what is defaultSearchIndexName and why is it required as parameter and not simply a class variable

Reply:
1. It's breaking changes of the v9 API. each enrollment should be indexed to be available in the search (i described that above and in my report). You could add different indexes (this allows to keep the single facemaps DB, but also to have a separate DB for each app - they just using different searchIndexName). defaultSearchIndexName is the index name used by GoodDollar app. it comes from the config/env

2. ZoomAPI class is a low-level API wrapping facetec server endpoints. I designed it to provide the most exact interface as the zoom server. So it consumes searchIndexName param. On the more high-level components (e.g. ZoomProvider) we're getting that param from the env/config and passing it down. It's single responsible rule of SOLID - one class one task. ZoomAPI works with API directly, provider provides some unified FV interface configured by the env (keepEnrollments, defaultSearchIndexName etc).